src/get_imports.rs

```diff
@@ -7,23 +7,36 @@ use swc_common::source_map::Pos;
 use swc_ecma_ast::Str;
 use swc_ecma_parser::Capturing;
 use swc_ecma_parser::{lexer::Lexer, Parser, StringInput, Syntax};
-pub struct SourceSpecifiers {
+
+use crate::error::GetImportError;
+struct SourceSpecifiers {
     specifiers: Vec<swc_ecma_ast::ImportSpecifier>,
     source: Str,
 }
 
 pub fn get_imports_map_from_file<'a>(
     file_path: &'a PathBuf,
-) -> HashMap<String, Option<HashSet<String>>> {
-    let imports = get_imports_from_file(&file_path);
-    get_imports_map(&imports, &file_path)
+) -> Result<HashMap<String, Option<HashSet<String>>>, GetImportError> {
+    if file_path.exists() {
+        let imports = match get_imports_from_file(&file_path) {
+            Ok(i) => i,
+            Err(e) => return Err(e),
+        };
+        return get_imports_map(&imports, &file_path);
+    }
+    Err(GetImportError::FileDoesNotExist(file_path.to_str().unwrap().to_string()))
 }
 
-fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Vec<SourceSpecifiers> {
+fn get_imports_from_file<'a>(file_path: &'a PathBuf) -> Result<Vec<SourceSpecifiers>, GetImportError> {
+    let path_string = match file_path.to_str() {
+        Some(path) => path,
+        None => return Err(GetImportError::ReadTsFileError(None)),
+    };
     let cm = Arc::<swc_common::SourceMap>::default();
-    let fm = cm
-        .load_file(Path::new(file_path.to_str().unwrap()))
-        .expect("Could not load file");
+    let fm = match cm.load_file(Path::new(path_string)) {
+        Ok(f) => f,
+        Err(_) => return Err(GetImportError::ReadTsFileError(Some(path_string.to_string()))),
+    };
     let handler = Handler::with_tty_emitter(ColorConfig::Auto, true, false, Some(cm.clone()));
```

Comment:
Errors should be collected into the return type instead of being output on stdout, I described how in another comment